### ISSUE TITLE
Supporting BatchNorm Layer

### ISSUE THREAD

**Issue by hthieu166** [APPFL/APPFL#94]
I would suggest the support of BatchNorm in APPFL. 
- `BatchNorm` layer has the `running_mean/running_var`, updated during training from training epochs and fixed during testing in `model.eval()` mode (check the docs at [BatchNorm2d](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html,), please pay additional attention to the track_running_stats argument).
- APPFL updates/initializes all model’s parameters listed under `model.named_parameters()`. Hence, the running mean/variance is NOT updated/synced at the server.
- Typically, these parameters should be updated. E.g., the function `load_state_dict()`  used when loading a model from a pre-trained weight also loads/updates these running estimated parameters.

**Comment by minseok-ryu** [comment]
Previously, ``model.named_parameters()`` has been used when updating the global model parameters.
In this revision, it has been replaced with ``self.model.state_dict()`` which includes ``running_mean/running_var``.
For parameters that belong to the ``model.named_parameters()``, we use the same global update before.
For parameters that do not belong to the  ``model.named_parameters()``, we take average on the local parameters to update the global parameters.

```
for name in self.model.state_dict():        
             if name in self.list_named_parameters: 
                 self.global_state[name] += self.step[name]            
             else:
                 tmpsum = torch.zeros_like(self.global_state[name], device=self.device)                
                 for i in range(self.num_clients):
                     tmpsum += self.primal_states[i][name]                
                 self.global_state[name] = torch.div(tmpsum, self.num_clients)
```
